Code/GraphMol/StructChecker/StructChecker.cpp

```diff
@@ -0,0 +1,176 @@
+//
+//  Copyright (C) 2016 Novartis Institutes for BioMedical Research
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include "StructChecker.h"
+#include "Pattern.h"
+#include "Stereo.h"
+#include "ReCharge.h"
+#include "Tautomer.h"
+#include "StripSmallFragments.h"
+
+namespace RDKit {
+namespace StructureCheck {
+
+unsigned StructChecker::checkMolStructure(RWMol &mol) const {
+  unsigned flags = NO_CHANGE;  // == 0. return value
+
+  if (0 != Options.MaxMolSize && (mol.getNumAtoms() > Options.MaxMolSize ||
+                                  mol.getNumBonds() > Options.MaxMolSize)) {
+    return SIZE_CHECK_FAILED;
+  }
+
+  if (mol.getNumAtoms() == 0) {
```

Comment:
Just a note for use while testing: STRUCHK does not give a SIZE_CHECK_FAILED error for zero-atom molecules, but I agree that it's good to do so. Still, this will lead to mis-matches.